adapters/utils/utils.ts

```diff
@@ -0,0 +1,54 @@
+import { Balances } from "@defillama/sdk"
+import { FetchResponseValue } from "../types";
+
+export function createBalanceFrom(chain: string, timestamp: number | undefined, amount: FetchResponseValue | undefined): Balances {
+  const balance = new Balances({ chain, timestamp })
+  if (amount) {
+    if (typeof amount === 'number' || typeof amount === 'string') {
+      balance.addUSDValue(amount)
+    } else {
+      balance.addBalances(amount)
+    }
+  }
+  return balance;
+}
+
+export async function subtractBalance(balance: Balances, amount: FetchResponseValue | undefined) {
+  if (amount) {
+    if (typeof amount === 'number' || typeof amount === 'string') {
+      const usdValue = await balance.getUSDValue()
+      balance.resizeBy(0)
+      balance.addUSDValue(Number(usdValue) - Number(amount))
+    } else {
+      balance.subtract(amount)
+    }
+  }
+}
+
+export async function validateAdapterResult(chain: string, result: any) {
+  // validate and auto compoute revenue if any
+  for (const key of ['daily', 'total']) {
+    const recordFees = `${key}Fees`
+    const recordSupplySideRevenue = `${key}SupplySideRevenue`
+    const recordProtocolRevenue = `${key}ProtocolRevenue`
+    const recordRevenue = `${key}Revenue`
+
+    if (result[recordFees]) {
```

Comment:
can you also add a check if we have label breakdown in the response? this way, we are certain that we are setting these default values only for the new adapters (yeah, it is bit of a hack, good to add why in the comment)